### ISSUE TITLE
Brave Cosmetic filtering script blocked on website due to CSP restriction

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#13929]
## Description 
Noticed this on GitHub tonite; nothing seemed to break webcompat-wise, but there is an error logged to F12 devtools/console
```
VM8:1 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src github.githubassets.com". Either the 'unsafe-inline' keyword, a hash ('sha256-AbpHGcgLb+kRsJGnwFEktk7uzpZOCcBY74+YBdrKVGs='), or a nonce ('nonce-...') is required to enable inline execution.

(anonymous) @ VM8:1
```

When you click the `VM8:1` and follow it,  the script shown is the cosmetic filtering script. Sites which have restrictive CSP may run into issues where cosmetic filtering may experience an issue

This may be an issue on 1.20 and newer (where cosmetic filtering landed with https://github.com/brave/brave-core/pull/6735)

cc: @pes10k 

## Steps to Reproduce

   1. Visit GitHub
   2. Use F12 or Shift+Cmd+i to open dev tools
   3. Check console tab for CSP error

## Reproduces how often: 
100% on GitHub; let's try to log more websites this happens on

**Comment by pes10k** [comment]
Just to add a but more detail, the new cosmetic filtering implementation seems to be injecting scripts into the page, which will cause errors on pages with restrictive inline CSP policies (since the cosmetic filtering script wont be nonce'd or similar),

cc @SergeyZhukovsky

**Comment by SergeyZhukovsky** [comment]
I'm going to check what we can do in that case

**Comment by stephendonner** [comment]
Verified `FIXED` using the inline testplan, with

Brave | 1.21.51 Chromium: 88.0.4324.152 (Official Build) beta (x86_64)
-- | --
Revision | 6579930fc53b4dc589c042bec9d0a3778326974d-refs/branch-heads/4324@{#2106}
OS | macOS Version 11.2.1 (Build 20D74)

Before fix (used `1.22.19 Chromium: 89.0.4389.40 (Official Build) nightly (x86_64)`:

![image](https://user-images.githubusercontent.com/387249/107691671-98bb3d00-6c60-11eb-8fd4-5cc541f11b7c.png)


After fix (used `1.21.51 Chromium: 88.0.4324.152 (Official Build) beta (x86_64)`:

![image](https://user-images.githubusercontent.com/387249/107691663-96f17980-6c60-11eb-94f5-1daaeae8115f.png)

Verified that the `Refused to execute inline script because it violates the following Content Security Policy directive: "script-src github.githubassets.com". Either the 'unsafe-inline' keyword, a hash ('sha256-AbpHGcgLb+kRsJGnwFEktk7uzpZOCcBY74+YBdrKVGs='), or a nonce ('nonce-...') is required to enable inline execution.` no longer shows up.

**Comment by srirambv** [comment]
Verification passed on Samsung Tab A with Android 10 running 1.21.70 x64 build
- Verified test plan from issue description
![image](https://user-images.githubusercontent.com/17010094/109466335-79dee800-7a61-11eb-92f1-267b8956d42b.png)
----------
Verification passed on OnePlus 6T with Android 10 running 1.21.70 x64 build
- Verified test plan from issue description
![image](https://user-images.githubusercontent.com/17010094/109466434-9d099780-7a61-11eb-9730-33d0677e31e5.png)
